src/firefly/js/visualize/ui/ExpandedOptionsPopup.jsx

```diff
@@ -133,59 +141,98 @@ function makeModel(tbl_id,plotViewAry, expandedIds, oldModel) {
 }
 
 function dialogComplete(tbl_id) {
-    const model= getTblById(tbl_id);
+    const model = getTblById(tbl_id);
     if (!model) return;
-    const plotIdAry= model.tableData.data.map( (d) => d[PID_IDX] );
+    const plotIdAry = model.tableData.data.map((d) => d[PID_IDX]);
     if (isEmpty(plotIdAry)) return;
 
-    const currentPlotIdAry= getViewerItemIds(getMultiViewRoot(),EXPANDED_MODE_RESERVED);
-    if (plotIdAry.join('')===currentPlotIdAry.join('')) return;
+    const currentPlotIdAry = getViewerItemIds(getMultiViewRoot(), EXPANDED_MODE_RESERVED);
+    if (plotIdAry.join('') === currentPlotIdAry.join('')) return;
     if (!plotIdAry.includes(visRoot().activePlotId)) {
         dispatchChangeActivePlotView(plotIdAry[0]);
     }
     dispatchReplaceViewerItems(EXPANDED_MODE_RESERVED, plotIdAry, IMAGE);
 }
 
-// const deleteFailedNEW= () => {
-//     getPlotViewAry(visRoot()).forEach( (pv) => {
-//         if (pv.serverCall==='fail') {
-//             dispatchDeletePlotView({plotId:pv.plotId}) ;
-//         }
-//     });
-// };
+const deleteFailedNEW = () => {
+    getPlotViewAry(visRoot()).forEach((pv) => {
+        if (pv.serverCall === 'fail') {
+            dispatchDeletePlotView({plotId: pv.plotId});
+        }
+    });
+};
+
+let selectedRows;
+let allSelected = false;
+
+const RemoveSelected = () => {
+    if (selectedRows) {
+        if (allSelected) {
+            getPlotViewAry(visRoot()).forEach((pv) => {
+                    dispatchDeletePlotView({plotId: pv.plotId});
+                }
+            );
+        } else {
+            let idx = 0;
+            getPlotViewAry(visRoot()).forEach((pv) => {
+                if (selectedRows.has(idx)) {
+                    dispatchDeletePlotView({plotId: pv.plotId});
+                }
+                idx++;
+            });
+            selectedRows.clear();
+        }
+    }
+}
```

Comment:
For what you want to do, there is no need to watch for table changes.  To get the selectInfo of a table, simply call:  `TableUtil.js:getTblInfoById`  i.e.

```
const selectInfoCls = SelectInfo.newInstance(getTblInfoById(TABLE_ID)?.selectInfo);
selectInfoCls.getSelected().forEach( (rowIdx) => {
    const plotId = convertRowIdxToPlotId(rowIdx);    // todo:  need to implement
    dispatchDeletePlotView({plotId});
});
```